include/CL/sycl/atomic.hpp

```diff
@@ -79,18 +79,34 @@ class atomic {
       memory_order::relaxed) volatile
   {
 #ifdef __HIPSYCL_DEVICE_CALLABLE__
+#ifdef SYCL_DEVICE_ONLY
     return atomicExch(_ptr, operand);
+#else
+    return __atomic_exchange_n(_ptr, operand, __ATOMIC_RELAXED);
```

Comment:
It's fine if we only support relaxed memory order for now, but could we maybe at least add `assert(memoryOrder == memory_order::relaxed)` on CPU? This way the user would notice that what he does is unsupported rather than the code just using a potentially different memory order.

Reply:
Sure. I'll fix this. Currently, the only `memory_order` that exists is `relaxed`, so the assertion is vacuous, but I get the point about making it future-proof.